src/ProjectOrigin.Vault/CommandHandlers/CheckForWithdrawnCertificatesCommandHandler.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using MassTransit;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using ProjectOrigin.Vault.Database;
+using ProjectOrigin.Vault.Models;
+using ProjectOrigin.Vault.Options;
+
+namespace ProjectOrigin.Vault.CommandHandlers;
+
+public record CheckForWithdrawnCertificatesCommand
+{
+}
+
+public class CheckForWithdrawnCertificatesCommandHandler : IConsumer<CheckForWithdrawnCertificatesCommand>
+{
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly IHttpClientFactory _httpClientFactory;
+    private readonly ILogger<CheckForWithdrawnCertificatesCommandHandler> _logger;
+    private readonly NetworkOptions _networkOptions;
+    public CheckForWithdrawnCertificatesCommandHandler(IUnitOfWork unitOfWork,
+        IOptions<NetworkOptions> networkOptions,
+        IHttpClientFactory httpClientFactory,
+        ILogger<CheckForWithdrawnCertificatesCommandHandler> logger)
+    {
+        _unitOfWork = unitOfWork;
+        _httpClientFactory = httpClientFactory;
+        _logger = logger;
+        _networkOptions = networkOptions.Value;
+    }
+
+    public async Task Consume(ConsumeContext<CheckForWithdrawnCertificatesCommand> context)
+    {
+        var stamps = _networkOptions.Stamps;
+
+        foreach (var stamp in stamps)
+        {
+            var cursors = await _unitOfWork.WithdrawnCursorRepository.GetWithdrawnCursors();
+            var matchingCursor = cursors.FirstOrDefault(x => x.StampName == stamp.Key) ?? new WithdrawnCursor
+            {
+                StampName = stamp.Key,
+                SyncPosition = 0,
```

Comment:
Why both a sync time and position ?

Reply:
sync time is just metadata for when last time we did a sync?

Reply:
Yes. It's so we can see on the table when the last time was we synced towards a given stamp. So call it extra info I guess